### DISCUSSION TITLE
compatibility of Parameter Backups when building from Master

### DISCUSSION THREAD

**Discussion by gitrimius** [shanteacontrols/OpenDeck#74] [Q&A]
Hi,
I  have a lot of pots, buttons etc defined in the web-interface. In the past I had to reconfigure everything after an update om my fw (build from master for a STM32 board). Could you explain a bit when/or not the parameter backups are compatible with future builds. Or what I should be aware of.

Thanks a lot!

**Comment by paradajz** [comment]
Factory reset occurs when I change the database layout in firmware. It also occurs if, for instance, you build default firmware for Black Pill board, then you create your own variant with different amount of pots, buttons etc. - in this case amount of components changes, therefore database layout changes.

**Reply by gitrimius** [reply]
ok, thanks.  I think I understand. 
Am I correct to think that (supposing nothing changed from firmware side and my yml side):
- if I _flash_ the firmware, I first need to do a Backup and then import the backup?
- If I _update_ (by using the web-interface) my settings should automatically be used?
I am asking because the last few time I tried the update via web-interface it did not reflect my changes, but flashing the corresponding hex file did work...
Thanks a lot for taking the time to help this beginner...
